WearOAuth/oauth-device-grant/src/main/java/com/example/android/wearable/oauth/devicegrant/AuthDeviceGrantActivity.kt

```diff
@@ -38,22 +55,45 @@ class AuthDeviceGrantActivity : ComponentActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_auth)
-        val viewModel by viewModels<AuthDeviceGrantViewModel>()
+        setContent { AuthenticateScreen(deviceGrantViewModel = viewModel()) }
+    }
+}
 
-        // Start the OAuth flow when the user presses the button
-        findViewById<View>(R.id.authenticateButton).setOnClickListener {
-            viewModel.startAuthFlow()
-        }
+@Composable
+fun AuthenticateScreen(deviceGrantViewModel: AuthDeviceGrantViewModel) {
+    val listState = rememberScalingLazyListState()
+    val uiState = deviceGrantViewModel.uiState.collectAsState()
+    val localContext = LocalContext.current
 
-        // Show current status on the screen
-        viewModel.status.observe(this) { statusText ->
-            findViewById<TextView>(R.id.status_text_view).text = resources.getText(statusText)
+    Scaffold(
```

Comment:
If you use AppScaffold + ScreenScaffold from Horologist you can remove the TimeText and PositionIndicator since those will be automatically added.
Also, feel free to remove the Vignette which is not recommended by UX anymore.